src/asm_writing/rewriter.cpp

```diff
@@ -1264,6 +1288,10 @@ void RewriterVar::refConsumed() {
     last_refconsumed_numuses = uses.size();
 }
 
+bool RewriterVar::needsDecref() {
+    return reftype == RefType::OWNED && !this->refHandedOff();
```

Comment:
I think it's possible for us to have multiple refs that we would need to remove from an object -- right now the rewriter emits all the increfs as early as possible, rather than emitting them just when they're needed.  So I think we either need to change the rewriter to emit the increfs on-demand, or to keep track of how many refs it's keeping at any given point (I think the first option sounds easier?).  Or we could just assert that it doesn't add multiple refs since that might not actually happen in practice :P